Energy_Preserving.jl

```diff
@@ -0,0 +1,524 @@
+    #we save the trees here
+    auxiliar = Array{Int64}(undef, m)
+    longitud = length(array)
+    auxiliar[m] = longitud
+    #the number of subtrees
+    n = m-1
+    subtrees = Vector{Any}(undef, n)
+    for i in 2:longitud
+        if array[i] == 2
+            contador += 1
+            auxiliar[contador] = i
+            if contador == n
+                break
+            end
+        end
+    end
+    for i in 1:n
+        subtrees[i] = array[auxiliar[i]:(auxiliar[i+1]-1)]
+    end
+    push!(subtrees[n],array[longitud])
+    return subtrees
+end
+         
+
+#generates all equivalent trees
+function equivalent_trees(array)
+    tree = rootedtree(array)
+    l = length(array)
+    #we get the permutations of the array
+    superarray = get_permutations(array)
+    lperm = length(superarray)
+    #eliminate all the arrays except for those for which the first element is equal to 1
+    for i in reverse(1:lperm) 
+        if superarray[i][1] != 1
+            splice!(superarray, i)
+        end
+    end
+    lperm = length(superarray)
+    #now, check if the levelsequence is the level sequence represents a RootedTree:
+    #the condition is that each number m cannot be followed by a number n such that
+    #n-m > 1. If that's the case, eliminate this array with 'splice!'
+    RTree_array = true
+    for i in reverse(1:lperm)
+        for j in 1:(l-1)
+            if (superarray[i][j+1] - superarray[i][j]) > 1
+                RTree_array = false
+            end
+        end
+        if RTree_array == false
+            splice!(superarray, i)
+        end
+        RTree_array = true
+    end
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        arbol = rootedtree(superarray[i])
+        isthesame_tree = arbol == tree
+        if isthesame_tree == false
+            splice!(superarray, i)
+        end
+    end
+    repeated_tree = false
+    lperm = length(superarray)
+    #we delete any repeated tree
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                repeated_tree = true
+            end
+        end 
+        if repeated_tree == true
+            splice!(superarray, i)
+        end
+        repeated_tree = false
+    end
+
+    return superarray
+end
+
+function get_permutations(arr)
+    perms = collect(permutations(arr))
+    return perms
+end
+
+function num_leafs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
+    l = length(arrays)
+    for i in 1:l
+        arrays[i] = vcat([1], arrays[i])
+    end
+    return arrays
+end
+
+#this function multplies the coefficient for its symmetry factor
+function symfact_normalization(coef,thetrees)
+    l = length(coef)
+    for i in 1:l
+        factor = 0
+        #because of the librarys we are using, some packages are repeated
+        #Then, we specify that the symmetry function comes from RootedTrees
+        factor = RootedTrees.symmetry(RootedTree(thetrees[i]))
```

Comment:
You mean to define `import RootedTrees; rt=RootedTrees` so that this line becomes `factor = rt.symmetry(RootedTree(thetrees[i]))`?

Reply:
No, the other way round. I guess Combinatorics.jl also exports a `symmetry` - which is the reason why you need to specify `RootedTrees.symmetry` here. What I suggest is to write
```julia
using Combinatorics: whatever, you_need, from_them
```
instead of
```julia
using Combinatorics
```
to import only the names you need from Combinatorics.jl.

Reply:
Okey, I got it.
Then this line becomes `factor = symmetry(RootedTree(thetrees[i]))`